parlai/agents/language_model/language_model.py

```diff
@@ -403,8 +425,8 @@ def predict(self, data, hidden, targets=None, is_training=True, y_lens=None):
             if bsz != self.batchsize:
                 self.hidden = self.model.init_hidden(bsz)
             loss = self.get_target_loss(data, self.hidden, targets, y_lens)
-            loss_dict['loss'] = loss
-            loss_dict['ppl'] = math.exp(loss)
+            self.metrics['loss'] += loss
+            self.metrics['num_tokens'] += sum(y_lens)
 
         return output, hidden, loss_dict, predictions
```

Comment:
seems like you can just get rid of `loss_dict` and not return it since it will always return `None`. You should also get rid of all of the `loss_dict` code in the `batch_act` function

Reply:
good catch